docs/gpu.ipynb

```diff
@@ -0,0 +1,344 @@
+{
```

Comment:
<p>Nice, looks good overall. I don't think it is particularly instructive for someone who isn't doing GPU themselves, but it shows a nice use case.</p><p> </p><p>How much struggle will it be to actually get this to work in the CI? getting llc and everything in might be painful.</p><p> </p><p>It would also be cool to see a rewrite, but I guess that is out of scope for the time being.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xdslproject/xdsl/pull/514/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Reply:
<p>Yes, this is meant as a working example, including the lowerings to binary, to get started. Not a GPU computing tutorial, which I think is out of scope of xDSL? At least out of this PR 🙂</p><p> </p><p>For the CI, getting llc and friends is trivial actually, I think the issues will more be around CUDA and GPU in docker from what I remember. Didn't try yet though.</p><p> </p><p>Do you mean implementing a rewrite about GPU in xDSL? I'll probably go with a lowering to it, depending on the SC paper needs mainly.</p>

Reply:
Yeah this currently is a bit out there and its is very much unclear why xDSL would be a benefit. But that can also just be a WIP. Lowering to this dialect from somewhere else will certainly be a good example as well.

This is mainly me suggesting things you could add, I am fine iwht it as is as well.

Reply:
You're totally right, in that sense it is a WIP, just printing the gpu dialect to MLIR's smarts.
The point I see is indeed using all their low-level optimizations and CUDA/ROCm compilation while writing higher level rewrite to it in xDSL.
I didn't check what they have as rewrites to GPU in MLIR yet though.